java/core/pom.xml

```diff
@@ -0,0 +1,194 @@
+                  <arg value="${test.proto.dir}/com/google/protobuf/non_nested_extension.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/non_nested_extension_lite.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test2.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test3.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_bad_identifiers.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_check_utf8.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_check_utf8_size.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_custom_options.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/any_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/field_presence_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_for_proto2_lite_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_for_proto2_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_initialization_order_test.proto" />
+                </exec>
+              </tasks>
+            </configuration>
+            <goals>
+              <goal>run</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Add the generated sources to the build -->
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <generatedSourcesDirectory>${generated.sources.dir}</generatedSourcesDirectory>
+          <generatedTestSourcesDirectory>${generated.testsources.dir}</generatedTestSourcesDirectory>
+        </configuration>
+      </plugin>
+
+      <!-- OSGI bundle configuration -->
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Bundle-DocURL>https://developers.google.com/protocol-buffers/</Bundle-DocURL>
+            <Bundle-SymbolicName>com.google.protobuf</Bundle-SymbolicName>
+            <Export-Package>com.google.protobuf;version=${project.version}</Export-Package>
+          </instructions>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <executions>
+
+          <!-- Configure a test jar artifact that can be used by other modules -->
+          <execution>
+            <id>test-jar</id>
```

Comment:
@xfxyjwf I suspect that the test jar would be deployed, which is somewhat unfortunate.  Assuming that's the case, we have a few options:

1. Just live with it. This may not be all that bad, since it allows us to keep the util `pom.xml` simple without having to re-generate the test sources from the `protobuf-java` module. It would also simply the configuration for new submodules that rely on the test protos.
2. We make the util project manually generate the test sources in the `protobuf-java` module. This would eliminate the test jar at the expense of complicating the util configuration (and potentially any new submodule that we add later that use the test protos).
3. We use gradle instead of maven.  This is ultimately the way we ended up going for grpc-java. Gradle is a lot more flexible here, since you can just depend on the test sources of another gradle project (e.g. https://github.com/grpc/grpc-java/blob/master/netty/build.gradle#L8)

Thoughts?